src/richie/apps/courses/templatetags/extra_tags.py

```diff
@@ -205,6 +208,61 @@ def render_tag(self, context, plugin, nodelist):
         return internal_context["content"]
 
 
+class PaginateNode(template.Node):
+    """
+    Renders a given template with its given context + the current pagination context
+
+    This is inspired by the dj-pagination "PaginateNode", with support for passing
+    specific context in addition to the paginate context.
+    """
+
+    def __init__(self, custom_template=None, additional_context=None):
+        self.template = custom_template
+        self.additional_context = additional_context or {}
+
+    def render(self, context):
+        template_context = paginate(context)
+        for name, value in self.additional_context.items():
+            template_context[name] = value.resolve(context)
+        return template.loader.render_to_string(self.template, template_context)
+
+
+@register.tag()
+def richie_paginate(parser, token):
```

Comment:
Maybe registering this tag as `paginate` to replace the dj-pagination one would be better (if possible)? This would prevent future usages of the dj-pagination tag that kinda becomes obsolete with this PR.